webapp/src/js/controllers/contacts.js

```diff
@@ -430,16 +434,28 @@ var _ = require('underscore'),
     var changeListener = Changes({
       key: 'contacts-list',
       callback: function(change) {
-        var limit = liveList.count();
+        const limit = liveList.count();
         if (change.deleted && change.doc.type !== 'data_record') {
           liveList.remove(change.doc);
         }
 
-        var withIds =
+        if (change.doc) {
+          liveList.invalidateCache(change.doc._id);
+          if (change.doc.fields) {
```

Comment:
I think this is implicitly checking if the changed doc was a report.  It might be helpful to either check `doc.type` here, or explain in a comment that this is what is happening.

Reply:
Even if I check the `doc.type` i would still need to do a null check for `doc.fields`, so I'm going with a comment here.